src/uct/ib/base/ib_device.c

```diff
@@ -303,6 +259,26 @@ static void uct_ib_async_event_handler(int fd, void *arg)
     ibv_ack_async_event(&event);
 }
 
+static void uct_ib_device_get_ids(uct_ib_device_t *dev)
+{
+    long vendor_id, device_id;
+
+    if ((ucs_read_file_number(&vendor_id, 1, UCT_IB_DEVICE_SYSFS_FMT,
+                              uct_ib_device_name(dev), "vendor") == UCS_OK) &&
+        (ucs_read_file_number(&device_id, 1, UCT_IB_DEVICE_SYSFS_FMT,
+                              uct_ib_device_name(dev), "device") == UCS_OK)) {
+        dev->pci_id.vendor = vendor_id;
+        dev->pci_id.device = device_id;
+        ucs_debug("%s vendor_id: 0x%x device_id: %d", uct_ib_device_name(dev),
+                  dev->pci_id.vendor, dev->pci_id.device);
+    } else {
```

Comment:
For systems like freebsd, which are not linux, does it make sense to fallback to standard device query ?

Reply:
standard device query returns different IDs (IB spec IDs, rather than PCI IDs)